contracts/application/versioning/ImplementationDirectory.sol

```diff
@@ -14,7 +14,7 @@ contract ImplementationDirectory is ImplementationProvider, Ownable {
    * @param contractName Name of the contract.
    * @param implementation Address of the added implementation.
    */
-  event ImplementationChanged(string contractName, address implementation);
+  event ImplementationChanged(string contractName, address indexed implementation);
```

Comment:
IMO the contract name should be indexed as well. See https://github.com/OpenZeppelin/openzeppelin-solidity/issues/992 for thoughts about the problem with indexing strings. Maybe we can adopt my proposal here, or do something similar.

Reply:
@frangio following our offline discussion, I opened a new issue to discuss this a bit more (https://github.com/zeppelinos/zos-lib/issues/199). Thanks